cmake/kokkos_pick_cxx_std.cmake

```diff
@@ -8,39 +8,12 @@ SET(KOKKOS_ENABLE_CXX14 OFF)
 SET(KOKKOS_ENABLE_CXX17 OFF)
 SET(KOKKOS_ENABLE_CXX20 OFF)
 IF (KOKKOS_CXX_STANDARD)
-  MESSAGE(DEPRECATION "Setting the variable Kokkos_CXX_STANDARD in configuration is deprecated - set CMAKE_CXX_STANDARD directly instead")
-
-  IF (${KOKKOS_CXX_STANDARD} STREQUAL "c++98")
-    MESSAGE(FATAL_ERROR "Kokkos no longer supports C++98 - minimum C++14")
-  ELSEIF (${KOKKOS_CXX_STANDARD} STREQUAL "c++11")
-    MESSAGE(FATAL_ERROR "Kokkos no longer supports C++11 - minimum C++14")
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++14")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++14'. Use '14' instead.")
-    SET(KOKKOS_CXX_STANDARD "14")
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++17")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++17'. Use '17' instead.")
-    SET(KOKKOS_CXX_STANDARD "17")
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++1y")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++1y'. Use '1Y' instead.")
-    SET(KOKKOS_CXX_STANDARD "1Y")
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++1z")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++1z'. Use '1Z' instead.")
-    SET(KOKKOS_CXX_STANDARD "1Z")
-  ELSEIF(${KOKKOS_CXX_STANDARD} STREQUAL "c++2a")
-    MESSAGE(WARNING "Deprecated Kokkos C++ standard set as 'c++2a'. Use '2A' instead.")
-    SET(KOKKOS_CXX_STANDARD "2A")
-  ENDIF()
+  MESSAGE(FATAL_ERROR "Setting the variable Kokkos_CXX_STANDARD in configuration is deprecated - set CMAKE_CXX_STANDARD directly instead")
 ENDIF()
 
-IF (NOT KOKKOS_CXX_STANDARD AND NOT CMAKE_CXX_STANDARD)
-  MESSAGE(STATUS "Setting default Kokkos CXX standard to 14")
+IF (NOT CMAKE_CXX_STANDARD)
   SET(KOKKOS_CXX_STANDARD "14")
-ELSEIF(NOT KOKKOS_CXX_STANDARD)
-  MESSAGE(STATUS "Setting default Kokkos CXX standard to ${CMAKE_CXX_STANDARD}")
+ELSE()
   SET(KOKKOS_CXX_STANDARD ${CMAKE_CXX_STANDARD})
 ENDIF()
-
-
-
-
-
+MESSAGE(STATUS "Setting default Kokkos CXX standard to ${KOKKOS_CXX_STANDARD}")
```

Comment:
The message is misplaced it should be in the `IF (NOT CMAKE_CXX_STANDARD)` branch

Reply:
We were printing in both cases before and I don't see a problem with that.

Reply:
If you insist, request changes.

Reply:
I viewed it in the same way as @masterleinad :  you have the message whatever the `CXX_STANDARD`